tests/test_delete_stmt.py

```diff
@@ -0,0 +1,268 @@
+    sa.Column('created_at', sa.DateTime),
+    sa.Column('updated_at', sa.DateTime)
+)
+
+product = sa.Table(
+    'products', meta,
+    sa.Column('id', sa.Integer, primary_key=True, autoincrement=False),
+    sa.Column('parent_id', sa.Integer),
+    sa.Column('name', sa.String(255)),
+    sa.Column('price', sa.Numeric(12, 4))
+)
+
+ham = sa.Table(
+    'ham', meta,
+    sa.Column('id', sa.Integer, primary_key=True, autoincrement=False)
+)
+
+spam = sa.Table(
+    'spam', meta,
+    sa.Column('id', sa.Integer, primary_key=True, autoincrement=False)
+)
+
+hammy_spam = sa.Table(
+    'ham, spam', meta,
+    sa.Column('ham_id', sa.Integer, sa.ForeignKey('ham.id')),
+    sa.Column('spam_id', sa.Integer, sa.ForeignKey('spam.id'))
+)
+
+
+def get_str(stmt):
+    return str(stmt.compile(dialect=RedshiftDialect()))
+
+
+def test_delete_stmt_nowhereclause():
+    del_stmt = sa.delete(customers)
+    assert get_str(del_stmt) == 'DELETE FROM customers'
+
+
+def test_delete_stmt_simplewhereclause1():
+    del_stmt = sa.delete(customers).where(
+        customers.c.email == 'test@test.test'
+    )
+    expeted = "DELETE FROM customers " \
+              "WHERE customers.email = %(email_1)s"
+    assert get_str(del_stmt) == expeted
+
+
+def test_delete_stmt_simplewhereclause2():
+    del_stmt = sa.delete(customers).where(
+        customers.c.email.endswith('test.com')
+    )
+    expected = "DELETE FROM customers " \
+               "WHERE customers.email " \
+               "LIKE '%%' || %(email_1)s"
+    assert get_str(del_stmt) == expected
+
+
+def test_delete_stmt_joinedwhereclause1():
+    del_stmt = sa.delete(orders).where(
+        orders.c.customer_id == customers.c.id
+    )
+    expected = "DELETE FROM orders " \
+               "USING customers " \
+               "WHERE orders.customer_id = customers.id"
+    assert get_str(del_stmt) == expected
+
+
+def test_delete_stmt_joinedwhereclause2():
+    del_stmt = sa.delete(
+        orders
+    ).where(
+        orders.c.customer_id == customers.c.id
+    ).where(
+        orders.c.id == items.c.order_id
+    ).where(
+        customers.c.email.endswith('test.com')
+    ).where(
+        items.c.name == 'test product'
+    )
+    expected = "DELETE FROM orders " \
+               "USING customers, items " \
+               "WHERE orders.customer_id = customers.id " \
+               "AND orders.id = items.order_id " \
+               "AND (customers.email LIKE '%%' || %(email_1)s) " \
+               "AND items.name = %(name_1)s"
+
+    assert get_str(del_stmt) == expected
+
+
+def test_delete_stmt_subqueryplusjoin():
+    del_stmt = sa.delete(
+        orders
+    ).where(
+        orders.c.customer_id.in_(
+            sa.select(
+                [customers.c.id]
+            ).where(customers.c.email.endswith('test.com'))
+        )
+    ).where(
+        orders.c.id == items.c.order_id
+    ).where(
+        items.c.name == 'test product'
+    )
+    expected = "DELETE FROM orders " \
```

Comment:
you can make these a docstring and use code like: https://github.com/graingert/redshift_sqlalchemy/blob/master/tests/test_unload_from_select.py#L8-L16

I don't like using "\" for line splitting